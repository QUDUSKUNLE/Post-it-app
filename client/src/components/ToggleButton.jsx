import React from 'react';
import $ from 'jquery';

/**
 * @export Footer component
 * @description - Footer Component
 * @return {object} ToggleButton component
 */
const ToggleButton = () =>
  (<div>
    <p className="visible-xs">
      <button
        type="button"
        className="btn btn-default btn-xs off-canvas"
        data-toggle="offcanvas"
        onClick={() => {
          if ($('.row-offcanvas').attr('class').indexOf('active') === -1) {
            $('.row-offcanvas').addClass('active');
          } else {
            $('.row-offcanvas').removeClass('active');
          }
        }}
      >
        <i className="glyphicon glyphicon-chevron-left"></i>
      </button>
    </p>
  </div>
  );

export default ToggleButton;
