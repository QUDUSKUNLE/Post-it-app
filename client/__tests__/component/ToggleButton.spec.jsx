import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import ToggleButton from '../../src/components/ToggleButton';

describe('<ToggleButton />', () => {
  it('component should be defined', () => {
    expect(ToggleButton).toBeDefined();
  });
  it('should render correctly', () => {
    const component = shallow(<ToggleButton />);
    expect(component).toMatchSnapshot();
    expect(component.find('button')).toHaveLength(1);
    expect(component.find('p')).toHaveLength(1);
  });
});
