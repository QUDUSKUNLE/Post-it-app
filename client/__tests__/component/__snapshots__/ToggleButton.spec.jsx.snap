// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton /> should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <p
      className="visible-xs"
    >
      <button
        className="btn btn-default btn-xs off-canvas"
        data-toggle="offcanvas"
        onClick={[Function]}
        type="button"
      >
        <i
          className="glyphicon glyphicon-chevron-left"
        />
      </button>
    </p>
  </div>,
  "nodes": Array [
    <div>
      <p
        className="visible-xs"
      >
        <button
          className="btn btn-default btn-xs off-canvas"
          data-toggle="offcanvas"
          onClick={[Function]}
          type="button"
        >
          <i
            className="glyphicon glyphicon-chevron-left"
          />
        </button>
      </p>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ToggleButton />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <p
            className="visible-xs"
          >
            <button
              className="btn btn-default btn-xs off-canvas"
              data-toggle="offcanvas"
              onClick={[Function]}
              type="button"
            >
              <i
                className="glyphicon glyphicon-chevron-left"
              />
            </button>
          </p>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <p
            className="visible-xs"
          >
            <button
              className="btn btn-default btn-xs off-canvas"
              data-toggle="offcanvas"
              onClick={[Function]}
              type="button"
            >
              <i
                className="glyphicon glyphicon-chevron-left"
              />
            </button>
          </p>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ToggleButton />,
}
`;
